### PR TITLE
Mda lite dashboard

### PR DESCRIPTION
**Pull request by trevlenb2**

### CODE REVIEW

File: src/main/java/com/revealprecision/revealserver/service/FormFieldService.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@RequiredArgsConstructor
+public class FormFieldService {
+
+  private final FormFieldRepository formFieldRepository;
+
+  public FormField findByNameAndFormTitle(String name, String formTitle) {
+    return formFieldRepository.findByNameAndFormTitle(name, formTitle);
+  }
+
+  public FormField createFormField(FormFieldRequest formFieldRequest) {
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/service/FormFieldService.java:21]: is it that uniqueness on (name, form_field) is handled at database level, but that failure is only visible on debug, maybe we throw not unique exception at api level.

File: src/main/java/com/revealprecision/revealserver/api/v1/controller/ReportDashboardController.java [hunk 2]
```diff
@@ -40,11 +40,12 @@ public List<String> getReportTypes(@PathVariable("identifier") UUID identifier)
     return ApplicableReportsEnum.valueOf(plan.getInterventionType().getCode()).getReportName();
   }
 
+
   @GetMapping("/reportData")
   public ResponseEntity<FeatureSetResponse> getDataForReports(
       @RequestParam(name = "reportType") String reportType,
       @RequestParam(name = "planIdentifier") UUID planIdentifier,
-      @RequestParam(name = "parentIdentifier", required = false) UUID parentIdentifier,
+      @RequestParam(name = "parentIdentifier", required = false) String parentIdentifier,
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/api/v1/controller/ReportDashboardController.java:48]: why are moving from uuid to string?

File: src/main/java/com/revealprecision/revealserver/api/v1/dto/factory/FormDataEntityTagEventFactory.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import com.revealprecision.revealserver.messaging.message.FormDataEntityTagEvent;
+import com.revealprecision.revealserver.messaging.message.FormDataEntityTagValueEvent;
+import com.revealprecision.revealserver.persistence.domain.Event;
+import com.revealprecision.revealserver.persistence.domain.Plan;
+import java.util.List;
+import java.util.UUID;
+
+public class FormDataEntityTagEventFactory {
+
+  // Proceed with caution here as new updates / removals to the object will prevent rewind of the streams application.
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/api/v1/dto/factory/FormDataEntityTagEventFactory.java:12]: maybe this comment should also be in the streams processing code or we make this code to not have that caution

File: src/main/java/com/revealprecision/revealserver/api/v1/dto/factory/PersonMetadataEventFactory.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import com.revealprecision.revealserver.messaging.message.PersonMetadataEvent;
+import com.revealprecision.revealserver.persistence.domain.Plan;
+import com.revealprecision.revealserver.persistence.domain.metadata.PersonMetadata;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class PersonMetadataEventFactory {
+
+  // Proceed with caution here as new updates / removals to the object will prevent rewind of the streams application.
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/api/v1/dto/factory/PersonMetadataEventFactory.java:12]: similar comment here to caution stream processor future developer

File: src/main/java/com/revealprecision/revealserver/api/v1/dto/response/LocationPropertyResponse.java [hunk 5]
```diff
@@ -33,7 +33,7 @@ public class LocationPropertyResponse {
   private Object numberOfChildrenTreated;
   private Object numberOfChildrenEligible;
   private Object sprayCoverage;
-  private UUID id;
+  private String id;
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/api/v1/dto/response/LocationPropertyResponse.java:36]: why the move from UUI to string?

File: src/main/java/com/revealprecision/revealserver/constants/FormConstants.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package com.revealprecision.revealserver.constants;
+
+public class FormConstants {
+  public static final String CDD_SUPERVISOR_DAILY_SUMMARY_HEALTH_WORKER_SUPERVISOR_FIELD = "health_worker_supervisor";
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/constants/FormConstants.java:4]: if this class will be used to hold constants, a suggestion would be to just change from class to interface and then wont have  to specify static final

File: src/main/java/com/revealprecision/revealserver/constants/KafkaConstants.java [hunk 7]
```diff
@@ -1,4 +1,4 @@
-package com.revealprecision.revealserver.messaging;
+package com.revealprecision.revealserver.constants;
 
 public class KafkaConstants {
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/constants/KafkaConstants.java:3]: similar suggestion as above around class vs interface on constants

File: src/main/java/com/revealprecision/revealserver/persistence/domain/metadata/infra/MetadataList.java [hunk 8]
```diff
@@ -2,11 +2,13 @@
 
 import java.io.Serializable;
 import java.util.List;
+import lombok.Data;
 import lombok.Getter;
 import lombok.Setter;
 
 @Setter
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/persistence/domain/metadata/infra/MetadataList.java:9]: I think we can remove these

File: src/main/java/com/revealprecision/revealserver/persistence/domain/metadata/infra/MetadataObj.java [hunk 9]
```diff
@@ -2,15 +2,22 @@
 
 import java.io.Serializable;
 import java.util.List;
+import java.util.UUID;
+import lombok.Data;
 import lombok.Getter;
 import lombok.Setter;
 
 @Setter
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/persistence/domain/metadata/infra/MetadataObj.java:10]: I think we can remove these

**Review by lindagcaba — COMMENTED** [review]
Looking good, just made a few comments and suggestions

File: src/main/java/com/revealprecision/revealserver/api/v1/controller/ReportDashboardController.java (hunk 2 above)
Comment (trevlenb2) [reply src/main/java/com/revealprecision/revealserver/api/v1/controller/ReportDashboardController.java:48]: here when traversing through the hierarchy of the dashboard....country, province etc....we eventually get to supervisor and cdd....these "parents" are no longer UUID's....their identifiers are now string

**Review by lindagcaba — APPROVED** [review]

**Pull request merged**